examples/delete_metric_descriptors.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+
+	"regexp"
+	"time"
+
+	monitoring "cloud.google.com/go/monitoring/apiv3/v2"
+	"cloud.google.com/go/monitoring/apiv3/v2/monitoringpb"
+	"google.golang.org/api/iterator"
+
+	"google.golang.org/api/option"
+)
+
+var (
+	cloudMonitoringEndpoint = flag.String("address", "monitoring.googleapis.com:443", "address of monitoring API")
+	resourceContainer       = flag.String("resource_container", "", "target resource container, ex. projects/test-project")
+	dryRun                  = flag.Bool("dry_run", false, "whether to dry run or not")
+)
+
+/*
+* To acquire Application Default Credentials, run:
+
+gcloud auth application-default login
+
+* One way to run this file is to initialize a go module.
+* For example, move this file into a new directory and run the following:
+
+go mod init example.com/m
+go mod tidy
+go run delete_metric_descriptors_timestamps.go -resource_container=projects/test-project
+
+*/
+
+func main() {
+	flag.Parse()
+	ctx := context.Background()
+
+	client, err := monitoring.NewMetricClient(ctx, option.WithEndpoint(*cloudMonitoringEndpoint))
+
+	if err != nil {
+		log.Fatalf("failed to build NewMetricClient for %s", *cloudMonitoringEndpoint)
+	}
+
+	it := client.ListMetricDescriptors(
+		ctx,
+		&monitoringpb.ListMetricDescriptorsRequest{
+			Name: *resourceContainer,
+		})
+
+	var deleted = 0
+	var numTotal = 0
+
+	for {
+		resp, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			log.Fatalf("Failed ListMetricDescriptors request: %v", err)
+		}
+		var metricType = resp.Type
+		match, err := regexp.MatchString("insert_regex_pattern_to_match", metricType)
```

Comment:
I know this should probably by a var up above but
1. I am not totally sure how to do it
2. I don't have the bandwidth to test it
3. I'm scared about putting anything in as a default value

Reply:
Sure, we can take it from here

Reply:
QQ: I guess acceptance criteria for this script is to delete metric descriptors from certain project that matches certain regex, right? I can write the script quickly so that it also asks for confirmation if we want

Reply:
For 3, we can have an empty string as the default and have the tool fail if it's empty. What do you think?

Reply:
@bwplotka yes - that's exactly it - to help people delete bad (statsd-style) descriptors from the system so they can free up their quota and speed up their autocomplete.

Reply:
We might want to expand it to accept an array of projects that it iterates through, as often times you make the same bad metric names across multiple projects.

Reply:
All done.